designs/initial.md

```diff
@@ -0,0 +1,56 @@
+## `@rest` directive
+
+#### arguments:
+- route: path to rest endpoint. This could be a path or a full url. If a path, add to the endpoint given on link creation or from the context
+- provides: a map of variables to url params
+- method: the HTTP method to send the request via (i.e GET, PUT, POST)
+- type: The GraphQL type this will return
+
+#### notes:
+Its important that the rest directive could be used at any depth in a query, but once it is used, nothing nested in it can be graphql data, it has to be from the rest link or other resource (like and @client directive)
+
+
+## `@export` directive
+
+#### arguments:
+- as: the string name to create this as a variable to be used down the selection set
+
+#### notes:
+this is the same semantics that it will be supported on the server, but when used in a rest link it would allow for using it for futher calls (i.e. waterfall requests)
+
+```js
+const QUERY = gql`
+  query RestData($email: String!) {
+    users @rest(route: '/users/email/:email', provides: { email: $email }, method: 'GET', type: 'User') {
+      id @export(as: "id")
+      firstName
+      lastName
+      friends @rest(route: '/friends/:id', provides: { id: $id }, type: '[User]') {
+        firstName
+        lastName
+      }
+    } 
+  }
+`;
+```
+
+
+## `createRestLink`
+
+#### arguments:
+- fetch: an implementation of `fetch` (see the http-link for api / warnings)
```

Comment:
Could this use global `fetch` when not provided?

Reply:
yeah for sure! I'll update to note that. The `http-link` usage of fetch argument is what I'd like to replicate and its optional there